docs/proposals/multi-arch-image-mgmt.md

```diff
@@ -0,0 +1,328 @@
+
+Ideally, if ORAS extends the ability to manage a multi-arch image by introducing create, update, and annotate a multi-arch image, the problems listed above can be resolved. Managing an image index for a multi-arch image should be as easy as playing Legos. 
+
+The proposed CLI commands for managing a multi-arch image are listed below. The detailed use case and subcommands are articulated in the CLI Spec section.
+
+- Create a multi-arch image using OCI image index: `oras manifest index create`
+- Update a multi-arch image using OCI image index: `oras manifest index update`
+- Inspect a multi-arch image using OCI image index: `oras manifest inspect`
+- Annotate a multi-arch image using OCI image index: `oras manifest create/update --annotation`
+- List platform information of tags in a repository: `oras repo tags`
+
+The proposal uses an OCI image layout as the local store to create a multi-arch image as an OCI image index locally, then push the multi-arch image to the remote registry with ORAS.  
+
+### User scenario and desired experience
+
+The sample workflow will be as following:
+
+1. Assume there are two arch-specific images tagged as _“v1-linux-amd64”_ and _“v1-linux-arm64” and have been pushed to an OCI image layout _layout-dir_. List the tags in the OCI image layout:
+
+```bash
+$ oras repo tags --oci-layout layout-dir 
+
+v1-linux-amd64 
+v1-linux-arm64 
+```
+
+2. Create a multi-arch image by composing two image manifests into an image index, tag it with “v1” and push the image index to an OCI image layout “layout-dir” automatically:  
+
+```
+$ oras manifest index create --oci-layout layout-dir:v1 v1-linux-amd64 v1-linux-arm64 
+
+Resolving  v1-linux-amd64 
+Resolved   v1-linux-amd64 
+Resolving  v1-linux-arm64 
+Resolved   v1-linux-arm64 
+Packed     edb5bc1f0b5c application/vnd.oci.image.index.v1+json 
+Pushed     [oci-layout] layout-dir:v1 
+Digest: sha256:edb5bc1f0b5c21e9321b34e50c92beae739250fb8840905 
+
+An image index has been created and pushed to layout-dir:v1 
+```
+
+3. Inspect the image index from the OCI image layout: 
+
+```bash
+$ oras manifest fetch --oci-layout layout-dir:v1 --pretty  
+
+{ 
+  "schemaVersion": 2, 
+  "mediaType": "application/vnd.oci.image.index.v1+json", 
+  "manifests": [ 
+    { 
+      "mediaType": "application/vnd.oci.image.manifest.v1+json", 
+      "digest": "sha256:42c524c48e0672568dbd2842d3a0cb34a", 
+      "size": 1239, 
+      "platform": { 
+        "architecture": "amd64", 
+        "os": "linux" 
+      } 
+    }, 
+    { 
+      "mediaType": "application/vnd.oci.image.manifest.v1+json", 
+      "digest": "sha256:965945e1a08031a63d5970c1da7c39af231c36e4c0a", 
+      "size": 1239, 
+      "platform": { 
+        "architecture": "arm64", 
+        "os": "linux" 
+      } 
+    } 
+  ] 
+} 
+```
+
+4. Add (or change) annotations to an image index in the OCI layout: 
+
+```bash
+$ oras manifest index update --oci-layout layout-dir:v1 --annotation “platform=multi-arch” 
```

Comment:
I don't think including `index update` with `--annotation` now is a good idea. Do we replace the original annotations? Or only add new ones? It's hard to settle down for now. A better option is to add update annotation support in the future when someone requests this.

Reply:
I think updating an existing annotation could be moved to the next release as it needs further discussion.